### PR TITLE
refactor: bars

### PR DESCRIPTION
**Pull request by snqb**
- ProgressBar2 -> AnimatedProgress
- bars/freshnses -> FreshnessBar
- removed: bar, progressbar
- BREAKING: revamp  FreshnessBar props
- BREAKING: revamp AnimatedProgress props

BREAKING: everything basically

requires changes to `hops`

Video:

https://user-images.githubusercontent.com/19791699/184220165-102a42dd-715a-415f-bac7-599ae63e7bd4.mp4

### CODE REVIEW

File: src/components/animated-progress/AnimatedProgress.tsx [hunk 1]
```diff
@@ -15,13 +15,13 @@ export interface ProgressBar2Props extends Options {
  * @param minimum - Optional Number between 0 and 1 indicating the minimum value of the progress bar. Defaults to 0.08.
  * @param variant - Optional string indicating the color of progressing bar. Defaults to 'perf.green'
  */
-const ProgressBar2: FC<ProgressBar2Props> = ({
+export const AnimatedProgress = ({
```
Comment (snqb) [inline src/components/animated-progress/AnimatedProgress.tsx:18]: I think the new name describes it well

File: src/components/bar/bar.stories.tsx [hunk 2]
```diff
@@ -1,114 +0,0 @@
...
-import { Story, Meta } from '@storybook/react/types-6-0';
-
-import Bar, { BarProps } from './index';
-import ProgressBar, { ProgressBarProps } from './progress';
-import FreshnessBar, { FreshnessBarProps } from './freshness';
-
-export default {
-  title: 'Quartz/Bars',
-  component: Bar,
-} as Meta;
```
Comment (snqb) [inline src/components/bar/bar.stories.tsx:11]: the `Bar` component was not used at all, as well as `progress`

File: src/components/freshness-bar/Bar.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+export const Bar = ({
+  variant = 'progress',
+  value,
+  width = '100px',
+  ...props
+}: Props) => (
+  <Box {...props} tx="variants.bar" variant={variant} sx={styles} width={width}>
+    <Box as="span" sx={overflowStyles(value)} />
+  </Box>
```
Comment (snqb) [inline src/components/freshness-bar/Bar.tsx:18]: just moved the old one here and put styles inside the file

File: src/components/freshness-bar/FreshnessBar.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import * as R from 'ramda';
+
+import Tooltip from '../tooltip';
+import { Bar } from './Bar';
+
+type BarSizes = 'sm' | 'md';
+
+export interface Props {
+  /** expiration date given as `Date`(e.g. via new Date(model.createdAt)) */
+  expirationDate: Date;
```
Comment (snqb) [inline src/components/freshness-bar/FreshnessBar.tsx:12]: now the old `time` prop is replaced with `expirationDate`, and it receives just `Date`
all of the usages in `hopsworks` were like `<FreshnessBar time={data.created.replace('T', ' ')} />`
now it is used it like `<FreshnessBar expirationDate={data.created} />`

previously component would manually remove `T` inside ISO-formatted string and do finicky stuff
just removed that crap and inserted a date picker in a storybook
image

File: src/components/freshness-bar/FreshnessBar.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  /** Size, short(sm) or medium (md) */
+  size?: BarSizes;
```
Comment (snqb) [inline src/components/freshness-bar/FreshnessBar.tsx:14]: old name was `length` and received `m` and `s`, we have no pattern like that in the project
now it's `size` and uses `md`, `sm` as we have in other components

File: src/components/freshness-bar/stories.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  },
+} as Meta;
+
+export const Default: Story<Props> = (props) => (
+  <Flex flexDirection="column" alignItems="center" sx={{ gap: '40px' }}>
+    <Flex sx={{ gap: '20px' }}>
+      <Label text="Half a year ago">
+        <FreshnessBar expirationDate={halfAYearAgo} />
+      </Label>
+      <Label text="3 months ago">
```
Comment (snqb) [inline src/components/freshness-bar/stories.tsx:36]: decided to provide some uneditable examples, as the component is not that intuitive imo

File: src/components/navigation/navigation.stories.tsx [hunk 7]
```diff
@@ -198,7 +198,7 @@ export const CustomTreeObject: Story<NavigationProps> = () => {
               component: (
                 <Box p="10px">
                   <h4>This is a custom component!</h4>
-                  <Bar value={0.99} />
```
Comment (snqb) [inline src/components/navigation/navigation.stories.tsx:201]: the only ever usage of `Bar` component ever :D

File: src/index.ts [hunk 8]
```diff
@@ -58,11 +58,7 @@ import TinyPopup from './components/popup/tiny';
 import usePopup from './utils/usePopup';
 
 // Bars
-import Bar from './components/bar';
-import ProgressBar from './components/bar/progress';
```
Comment (snqb) [inline src/index.ts:62]: these are goners, not used anywhere

File: src/components/freshness-bar/stories.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    expirationDate: new Date(monthAgo),
+    size: 'md',
+  },
+  argTypes: {
+    expirationDate: {
+      control: {
+        type: 'date',
+      },
+    },
+  },
```
Comment (esonhey) [inline src/components/freshness-bar/stories.tsx:27]: the addon-docs doesn't pick the property type correctly in case of Date?

**Review by esonhey — APPROVED** [review]
Good job Nice clean-ups!

File: src/components/freshness-bar/stories.tsx (hunk 9 above)
Comment (snqb) [reply src/components/freshness-bar/stories.tsx:27]: yeah, it gives a timestamp input
image

**Pull request merged**